File: app/src/main/java/com/cookiegames/smartcookie/activity/SettingsActivity.java
package com.cookiegames.smartcookie.activity;


import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.cookiegames.smartcookie.R;
import com.cookiegames.smartcookie.utils.PreferenceHelper;
import com.cookiegames.smartcookie.utils.ThemeUtils;

/**
 * Copyright (c) 2016 Vlad Todosin
 */
public class SettingsActivity extends AppCompatActivity{
    private MenuItem itm;
    @Override
    public void onCreate(Bundle savedInstanceState) {
        ThemeUtils theme = new ThemeUtils(this);
        theme.setTheme();
        super.onCreate(savedInstanceState);
        setContentView(R.layout.preference);
        Toolbar bar = (Toolbar)findViewById(R.id.settingsbar);
        setSupportActionBar(bar);
        setTitle(getResources().getString(R.string.action_settings));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getFragmentManager().beginTransaction().replace(R.id.layout, new SettingsFragment()).commit();

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == android.R.id.home) {
            onBackPressed();
            return true;
        }

        return false;
    }
    @Override
    public void onBackPressed() {
        if(PreferenceHelper.getIsBrowser() || PreferenceHelper.getIsLook()){
            setTitle(getResources().getString(R.string.action_settings));
            getFragmentManager().beginTransaction().replace(R.id.layout, new SettingsFragment()).commit();
            PreferenceHelper.setIsBrowserScreen(false);
            PreferenceHelper.setIsLookScreen(false);
        }
        else{
            Intent in = new Intent(this,MainActivity.class);
            in.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(in);
        }
    }
}

File: app/src/main/java/com/cookiegames/smartcookie/settings/activity/SettingsActivity.kt
/*
 * Copyright 2014 A.C.R. Development
 */
package com.cookiegames.smartcookie.settings.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.cookiegames.smartcookie.AppTheme
import com.cookiegames.smartcookie.R
import com.cookiegames.smartcookie.di.injector
import com.cookiegames.smartcookie.preference.UserPreferences
import com.cookiegames.smartcookie.settings.fragment.SettingsFragment
import javax.inject.Inject


class SettingsActivity : AppCompatActivity() {

    @Inject
    internal lateinit var userPreferences: UserPreferences


    override fun onCreate(savedInstanceState: Bundle?) {
        injector.inject(this)

        // set the theme
        when (userPreferences.useTheme) {
            AppTheme.LIGHT -> {
                setTheme(R.style.Theme_SettingsTheme)
            }
            AppTheme.DARK -> {
                setTheme(R.style.Theme_SettingsTheme_Dark)
            }
            AppTheme.BLACK -> {
                setTheme(R.style.Theme_SettingsTheme_Black)
            }
        }

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        val toolbar: Toolbar = findViewById(R.id.toolbar)

        try {
            setSupportActionBar(toolbar)
            val actionBar: ActionBar? = getSupportActionBar()
            if (actionBar != null) {
                actionBar.setDisplayShowTitleEnabled(false)
                actionBar.setDisplayHomeAsUpEnabled(true)
                actionBar.setDisplayShowHomeEnabled(true)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        toolbar.title = getString(R.string.settings)
        getSupportFragmentManager().beginTransaction().replace(R.id.container, SettingsFragment()).commit()
        overridePendingTransition(R.anim.slide_in_from_right, R.anim.fade_out_scale)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.getItemId()) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
